crmsh/bootstrap.py

```diff
@@ -932,31 +939,48 @@ def append_to_remote_file(fromfile, remote_node, tofile):
 
 
 def init_csync2():
+    host_list = _context.node_list_in_cluster
+
     logger.info("Configuring csync2")
     if os.path.exists(CSYNC2_KEY):
         if not confirm("csync2 is already configured - overwrite?"):
             return
 
     invoke("rm", "-f", CSYNC2_KEY)
-    with logger_utils.status_long("Generating csync2 shared key (this may take a while)"):
-        if not invokerc("csync2", "-k", CSYNC2_KEY):
-            utils.fatal("Can't create csync2 key {}".format(CSYNC2_KEY))
+    logger.debug("Generating csync2 shared key")
+    if not invokerc("csync2", "-k", CSYNC2_KEY):
+        utils.fatal("Can't create csync2 key {}".format(CSYNC2_KEY))
 
     csync2_file_list = ""
     for f in FILES_TO_SYNC:
         csync2_file_list += "include {};\n".format(f)
 
+    host_str = ""
+    for host in host_list:
+        host_str += 'host {};\n'.format(host)
+
     utils.str2file("""group ha_group
 {{
 key /etc/csync2/key_hagroup;
-host {};
+{}
 {}
 }}
-    """.format(utils.this_node(), csync2_file_list), CSYNC2_CFG)
+    """.format(host_str, csync2_file_list), CSYNC2_CFG)
+
+    if _context.skip_csync2:
+        for f in [CSYNC2_CFG, CSYNC2_KEY]:
+            sync_file(f)
 
-    utils.start_service("csync2.socket", enable=True)
-    with logger_utils.status_long("csync2 checking files"):
-        invoke("csync2", "-cr", "/")
+    for host in host_list:
+        logger.info("Starting {} service on {}".format(CSYNC2_SERVICE, host))
+        utils.start_service(CSYNC2_SERVICE, enable=True, remote_addr=host)
+
+    _msg = "syncing" if _context.skip_csync2 else "checking"
+    with logger_utils.status_long("csync2 {} files".format(_msg)):
+        if _context.skip_csync2:
+            csync2_update("/")
```

Comment:
Why is `csync2_update` called when it is "skipped"?

Reply:
When running `crm cluster init -x -y` then `crm cluster join`, above piece of codes will not be executed, the csync2.socket will not started and enabled
Then to configure and enable csync2, need to run `crm cluster init csync2 -y`, above piece of codes will be executed.
Since on that time, csync2 db need to be initialized, so need to run `csync2_update`